components/AccountingCategorySelect.tsx

```diff
@@ -314,15 +326,66 @@ const AccountingCategorySelect = ({
       onChange(newCategory);
     }
   };
+
+  const previousAccount = usePrevious(account);
+
+  const query = useQuery<AccountingCategorySelectQuery, AccountingCategorySelectQueryVariables>(
+    gql`
+      query AccountingCategorySelect($hostSlug: String!, $account: AccountReferenceInput) {
+        host(slug: $hostSlug) {
+          accountingCategories(account: $account) {
+            nodes {
+              ...AccountingCategoryFields
+            }
+          }
+        }
+      }
+      ${accountingCategoryFields}
+    `,
+    {
+      context: API_V2_CONTEXT,
+      variables: {
+        hostSlug: host.slug,
+        account: account?.slug ? { slug: account.slug } : null,
+      },
+      skip: !isOpen,
+    },
+  );
```

Comment:
If I understand this correctly, we're going to re-fetch accounting categories for each account. While this is fine for the expense page, it won't play well with one of the main use cases for this component: in the host dashboard > expenses list. For each account, we'll have to wait for the list to load, and incrementally increase the Apollo cache size with data we didn't really need to refetch.

As an alternative approach, I'd suggest to directly fetch `appliesTo` in the places that use this component and filtering from the frontend - like we already do for `expenseType`. Since we already have the host and the account, all the changes to this component could then fit in a very simple change to `getOptions`:

```es6
const isHostBudget = collective.host.id === (collective.parent.id ?? collective.id);
const expectedAppliesTo = isHostBudget ? 'HOST' : 'HOSTED_COLLECTIVES';
remove(categories, category => category.appliesTo !== expectedAppliesTo);
```

Reply:
> The expense list will not cause a fetch for every account in the list as the accounting category is only fetched when the picker is open. 

A common workflow for the categorization is to open the expenses list, filter it, and then set the category on each expense one by one, directly from the list.

> the cost would be negligible.

In terms of API load, yes. In terms of UX, with a slow connection, having to wait for the select to load each time you want to categorize an expense in a list could be a frustrating experience.

> Do we want to keep the account query filter? I think its nice to remove some biz logic from the frontend

I overall agree with keeping the business logic in the API as much as we can, but here the tiny logic (checking whether an account is the host of a hosted account) required seems worth it regarding the API <> frontend back-and-forth that we're able to cut by implementing it.